Sources/Client/Model/Channel/ChannelResponse.swift

```diff
@@ -54,8 +54,41 @@ public struct ChannelResponse: Decodable {
         if let lastMessage = messages.last,
             let messageRead = messageReads.first(where: { $0.user.isCurrent }),
             lastMessage.updated > messageRead.lastReadDate {
-            unreadMessageRead = messageRead
+            channel.unreadMessageRead = messageRead
+        }
+    }
+    
+    func calculateChannelUnreadCount() {
+        channel.unreadCountAtomic.set(0)
+        channel.mentionedUnreadCountAtomic.set(0)
+        
+        if messages.isEmpty {
+            return
         }
+        
+        var count = 0
+        var mentionedCount = 0
+        let currentUser = Client.shared.user
+        
+        if let unreadMessageRead = channel.unreadMessageRead {
+            for message in messages.reversed() {
```

Comment:
Why not, instead of this `for` loop:
```
let newMessages = messages.filter { $0.created > unreadMessageRead.lastReadDate }
count = newMessages.count
mentionedCount = newMessages.filter { $0.user != currentUser && $0.mentionedUsers.contains(currentUser) }.count // This is O(n^2), same with the current approach
```
I believe it's easier to read